### ISSUE TITLE
Syntax Error in controller.py

### ISSUE THREAD

**Issue by Synocode** [ianhi/mpl-interactions#239]
Getting a "TypeError: 'builtin_function_or_method' object is not subscriptable" from the linked line.

https://github.com/ianhi/mpl-interactions/blob/4ecd1b589abab82e983b595dc8448c818680d9ce/mpl_interactions/controller.py#L396

**Comment by ianhi** [comment]
Indeed that's a bug! Thanks for the report. Can you post the code you ran that resulted in that error so I can add it to the tests to prevent this in the future?

**Comment by ianhi** [comment]
(also you're welcome to open a PR if you want :) )

**Comment by ianhi** [comment]
ah I can replicate with this:
```python
x = np.random.randn(20)
y = 2 * x + np.random.randn(20) / 2

x2 = np.random.randn(20)
y2 = -2 * x2 - np.random.randn(20) / 2


fig, ax = plt.subplots()
ctrls = iplt.scatter(x, y, s=(10, 1000))
with ctrls:
    _ = iplt.scatter(x2, y2, s=ctrls["s"])
```

**Comment by Synocode** [comment]
Here's a minimal example which doesn't work for me and is close to how I encountered the issue.

```python
x = np.linspace(0, 2 * np.pi, 1000)
tau = np.linspace(5, 10)
beta = np.linspace(0.25, 1)
delta = np.linspace(0, 0.75)
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(10, 5))

def f1(tau, beta):
    return tau*np.arange(9).reshape((3,3)) + beta


def f2(tau):
    return  tau*np.arange(9).reshape((3,3))

controls = iplt.imshow(f1, tau=tau, beta=beta, ax=ax1)

with controls["tau"]:
    iplt.imshow(f2, ax=ax2)
```

**Comment by ianhi** [comment]
Should be fixed in version `0.20.2` which will be on pypi as soon as this finishes: https://github.com/ianhi/mpl-interactions/actions/runs/2000408310

thanks again for reporting!

**Comment by Synocode** [comment]
Awesome. Thanks for the quick turnaround!
